### PR TITLE
Add Chalice Support

### PR DESCRIPTION
**Pull request by joshajohnson**
## Description
Adds support for [Chalice](https://customkbd.com/products/chalice-pre-order)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/chalice/config.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN D3
+
+#ifdef RGBLIGHT_ENABLE
+    #define RGBLED_NUM 14
+    #define RGBLIGHT_SLEEP  
+    #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/chalice/config.h:43]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/chalice/info.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
+{
+    "keyboard_name": "Chalice",
+    "url": "https://customkbd.com/products/chalice-pre-order",
+    "maintainer": "CustomKBD",
+    "width": 20,
+    "height": 5.25,
```
Comment (drashna) [inline keyboards/chalice/info.json:6]:
```suggestion
```

File: keyboards/chalice/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "chalice.h"
```
Comment (drashna) [inline keyboards/chalice/keymaps/default/keymap.c:18]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/chalice/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/chalice/keymaps/default/keymap.c:20]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/chalice/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "chalice.h"
```
Comment (drashna) [inline keyboards/chalice/keymaps/via/keymap.c:18]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/chalice/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/chalice/keymaps/via/keymap.c:20]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/chalice/readme.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Chalice
+
+A Cheap Alice, powered by a Pro-Micro
+
+* Keyboard Maintainer: [CustomKBD](https://github.com/customkbd/)
+* Hardware Supported: Pro-Micro (or equivalent - Sea Micro, Elite-C etc.)
+* Hardware Availability: [CustomKBD](https://customkbd.com/products/chalice-pre-order)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/chalice/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/chalice/rules.mk [hunk 8]
```diff
@@ -0,0 +1,15 @@
+MCU = atmega32u4
+BOOTLOADER = atmel-dfu
+
+# Build Options
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/chalice/rules.mk:5]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by joshajohnson** [comment]
Thanks for the review Drashna :smile:

**Review by drashna — APPROVED** [review]

File: keyboards/chalice/readme.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+    make chalice:default:flash
```
Comment (fauxpark) [inline keyboards/chalice/readme.md:11]:
```suggestion
    make chalice:default

Flashing example for this keyboard:

    make chalice:default:flash
```

File: keyboards/chalice/rules.mk [hunk 10]
```diff
@@ -0,0 +1,15 @@
+MCU = atmega32u4
+BOOTLOADER = atmel-dfu
+
+# Build Options
```
Comment (fauxpark) [inline keyboards/chalice/rules.mk:4]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
```

File: keyboards/chalice/rules.mk [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/chalice/rules.mk:11]:
```suggestion
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
```

File: keyboards/chalice/rules.mk [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/chalice/rules.mk:14]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by joshajohnson** [comment]
Thanks for the feedback Fauxpark :)

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**